lib/engine-universal/src/link.rs

```diff
@@ -168,6 +168,17 @@ fn apply_relocation(
 
 /// Links a module, patching the allocated functions with the
 /// required relocations and jump tables.
+#[cfg_attr(
+    feature = "tracing",
+    tracing::instrument(skip(
+        allocated_functions,
+        jt_offsets,
+        function_relocations,
+        allocated_sections,
+        section_relocations,
+        trampolines
```

Comment:
Hmm I think I originally wanted to keep some info, and eventually gave up and skipped everything — that said the PR is not ready for in-depth review yet, sorry for having forgotten to open it as draft, I just published to get feedback about whether other people think tracing should be an optional or mandatory dependency :sweat_smile:

Reply:
Got it. My feeling is that there are very little we can hope to include as a field in spans today, so to me it makes most sense if we approach it from the direction of adding a field when we realize a need for it, rather than the other way around.

Introducing significant perf issues by accidentally debug-outputting parts of a 4MiB module structure seems like it'd be much worse than just missing that information, especially given that in production the contract can always be copied out from the database and investigated locally anyway.